### ISSUE TITLE
when I train the net, I meet an error. I'm new

### ISSUE THREAD

**Issue by xiaomimo** [unsky/FPN-caffe#21]
I0103 17:59:46.463167 29140 layer_factory.hpp:77] Creating layer fpn_loss_cls
I0103 17:59:46.463174 29140 net.cpp:106] Creating Layer fpn_loss_cls
I0103 17:59:46.463178 29140 net.cpp:454] fpn_loss_cls <- rpn_cls_score_reshape
I0103 17:59:46.463183 29140 net.cpp:454] fpn_loss_cls <- rpn_labels
I0103 17:59:46.463191 29140 net.cpp:411] fpn_loss_cls -> FPNClsLoss
I0103 17:59:46.463199 29140 layer_factory.hpp:77] Creating layer fpn_loss_cls
F0103 17:59:46.465251 29140 softmax_loss_layer.cpp:47] Check failed: outer_num_ * inner_num_ == bottom[1]->count() (245520 vs. 552960) Number of labels must match number of predictions; e.g., if softmax axis == 1 and prediction shape is (N, C, H, W), label count (number of labels) must be N*H*W, with integer values in {0, 1, ..., C-1}.
*** Check failure stack trace: ***

Thank you

**Comment by unsky** [comment]
use the new codes and compile the caffe-fpn in the codes

**Comment by xiaomimo** [comment]
@unsky Thank your answer. I have used the new codes but there is another error.

I0103 19:39:44.953022  4457 layer_factory.hpp:77] Creating layer loss_cls
I0103 19:39:44.953043  4457 net.cpp:106] Creating Layer loss_cls
I0103 19:39:44.953058  4457 net.cpp:454] loss_cls <- cls_score
I0103 19:39:44.953071  4457 net.cpp:454] loss_cls <- labels
I0103 19:39:44.953088  4457 net.cpp:411] loss_cls -> RcnnLossCls
I0103 19:39:44.953107  4457 layer_factory.hpp:77] Creating layer loss_cls
F0103 19:39:44.953361  4457 loss_layer.cpp:19] Check failed: bottom[0]->num() == bottom[1]->num() (4 vs. 1) The data and label should have the same number.
*** Check failure stack trace: ***


Thank you very much.

**Comment by xiaomimo** [comment]
@unsky Need to use your caffe-fpn and can not use faster rcnn's caffe?

**Comment by unsky** [comment]
faster rcnn-caffe is very old, the loss layer cannot infershape.

**Comment by xiaomimo** [comment]
@unsky There is no problems. Thank you very much!

**Comment by LeonJWH** [comment]
@unsky what have you changed for softmaxwithloss
